pkg/hydrate/controller_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hydrate
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"kpt.dev/configsync/pkg/api/configsync/v1beta1"
+	"kpt.dev/configsync/pkg/importer/filesystem/cmpath"
+	ft "kpt.dev/configsync/pkg/importer/filesystem/filesystemtest"
+)
+
+var originCommit = "1234567890abcdef"
+var differentCommit = "abcdef1234567890"
+
+func TestCheckCommitChanged(t *testing.T) {
+	testCases := []struct {
+		name          string
+		commitChanged bool
+		sourceCommit  string
+	}{
+		{
+			name:          "Run hydrate when source commit is unchanged",
+			commitChanged: false,
+			sourceCommit:  originCommit,
+		},
+		{
+			name:          "Run hydrate when souce commit is changed",
+			commitChanged: true,
+			sourceCommit:  differentCommit,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// create a temporary directory with a commit hash
+			tempDir, err := ioutil.TempDir(os.TempDir(), "run-hydrate-test")
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer func(path string) {
+				err := os.RemoveAll(path)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}(tempDir)
+			absTempDir := cmpath.Absolute(tempDir)
+
+			commitDir := filepath.Join(tempDir, originCommit)
+			err = os.Mkdir(commitDir, os.ModePerm)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// create a symlink to point to the temporary directory
+			dir := ft.NewTestDir(t)
+			symDir := dir.Root().Join(cmpath.RelativeSlash("test-hydrate-symlink"))
+			err = os.Symlink(commitDir, symDir.OSPath())
```

Comment:
add a `defer` to clean up the symlink after the test

Reply:
Done